Source/Events/CouldNotDeserializeEvent.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Dolittle.Artifacts;
+
+namespace Dolittle.Events
+{
+    /// <summary>
+    /// Exception that gets thrown when an <see cref="IEvent"/> couldn't be deserialized.
+    /// </summary>s
+    public class CouldNotDeserializeEvent : Exception
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CouldNotDeserializeEvent"/> class.
+        /// </summary>
+        /// <param name="artifact">The <see cref="ArtifactId"/> of the Artifact.</param>
+        /// <param name="eventType">The Type of the event.</param>
+        /// <param name="content">The content of the event.</param>
+        /// <param name="sequenceNumber">The Events position in the event log.</param>
+        /// <param name="ex">The catched exception which made deserialization impossible in the first place.</param>
+        public CouldNotDeserializeEvent(
+            ArtifactId artifact,
+            Type eventType,
+            string content,
+            ulong sequenceNumber,
```

Comment:
I think that we should use the Concepts here
```suggestion
            EventLogSequenceNumber sequenceNumber,
```

Reply:
The value given to the exception comes from the `Contracts.CommittedEvent.EventLogSequenceNumber` which is a `ulong`, so it doesn't have any `To<EventLogSequenceNumber>()` functions like the `Uuid`'s.

Reply:
Yes, you’re right